src/PKSim.Core/Mappers/MoleculeToQueryExpressionSettingsMapper.cs

```diff
@@ -28,6 +30,17 @@ public QueryExpressionSettings MapFrom(IndividualMolecule molecule, ISimulationS
          return new QueryExpressionSettings(expressionContainer, molecule.QueryConfiguration, moleculeName);
       }
 
+      public QueryExpressionSettings MapFrom(ExpressionProfileBuildingBlock expressionProfileBuildingBlock)
+      {
+         var containerGroups = expressionProfileBuildingBlock.
+            Where(x => x.Value.HasValue && x.HasExpressionName()).
+            Select(x => (categoryName:x.ContainerNameForRelativeExpressionParameter(), expressionValue:x.Value.Value)).ToList();
```

Comment:
let's call this container and not category?

Reply:
and why don't you create a method that reurnd an ExceprresionContainerInfo directly instead of loopng twice in the next line?